components/team/EditTeam.tsx

```diff
@@ -0,0 +1,297 @@
+import React, { useEffect, useState, useRef } from "react";
+import { ArrowLeftIcon, PlusIcon, TrashIcon } from "@heroicons/react/outline";
+import ErrorAlert from "@components/ui/alerts/Error";
+import { UsernameInput } from "@components/ui/UsernameInput";
+import MemberList from "./MemberList";
+import Avatar from "@components/Avatar";
+import ImageUploader from "@components/ImageUploader";
+import { Dialog, DialogTrigger } from "@components/Dialog";
+import ConfirmationDialogContent from "@components/dialog/ConfirmationDialogContent";
+import Modal from "@components/Modal";
+import MemberInvitationModal from "@components/team/MemberInvitationModal";
+import Button from "@components/ui/Button";
+import { Member } from "@lib/member";
+import { Team } from "@lib/team";
+
+export default function EditTeam(props: { team: Team | undefined | null; onCloseEdit: () => void }) {
+  const [members, setMembers] = useState([]);
+
+  const nameRef = useRef<HTMLInputElement>() as React.MutableRefObject<HTMLInputElement>;
+  const teamUrlRef = useRef<HTMLInputElement>() as React.MutableRefObject<HTMLInputElement>;
+  const descriptionRef = useRef<HTMLTextAreaElement>() as React.MutableRefObject<HTMLTextAreaElement>;
+  const hideBrandingRef = useRef<HTMLInputElement>() as React.MutableRefObject<HTMLInputElement>;
+  const logoRef = useRef<HTMLInputElement>() as React.MutableRefObject<HTMLInputElement>;
+  const [hasErrors, setHasErrors] = useState(false);
+  const [successModalOpen, setSuccessModalOpen] = useState(false);
+  const [showMemberInvitationModal, setShowMemberInvitationModal] = useState(false);
+  const [inviteModalTeam, setInviteModalTeam] = useState<Team | null | undefined>();
```

Comment:
I personally prefer using search params when modals are open. Subjective opinion but I like when things can be linked

https://github.com/calendso/calendso/blob/8e9703545af208aa073783a9f0643e635943e035/pages/event-types/index.tsx#L46
https://github.com/calendso/calendso/blob/8e9703545af208aa073783a9f0643e635943e035/pages/event-types/index.tsx#L88-L93